server/src/service.rs

```diff
@@ -300,4 +319,51 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn set_locked(
+        &self,
+        locked: bool,
+        objects: &[OwnedObjectPath],
+    ) -> Result<(Vec<OwnedObjectPath>, Vec<OwnedObjectPath>), ServiceError> {
+        let mut locked_changed = Vec::new();
+        let mut locked_unchanged = Vec::new();
+        let collections = self.collections.lock().await;
+
+        for object in objects {
+            for collection in collections.iter() {
+                if object == collection.path() {
+                    if locked == collection.is_locked().await {
+                        tracing::debug!(
+                            "Collection: {} is already {}.",
+                            object,
+                            if locked { "locked" } else { "unlocked" }
+                        );
+                        locked_changed.push(object.clone());
+                    } else {
+                        locked_unchanged.push(object.clone());
+                    }
+                    break;
+                } else if let Some(item) = collection.item_from_path(object).await {
+                    if locked == item.is_locked().await {
+                        tracing::debug!(
+                            "Item: {} is already {}.",
+                            object,
+                            if locked { "locked" } else { "unlocked" }
+                        );
+                        locked_changed.push(object.clone());
+                    // If the collection is in a similar state, update locked of
+                    // an item without a prompt.
```

Comment:
`// If the collection is in a similar state, update the Item without a prompt.` without extra verbosity